src/Common/MemoryTracker.cpp

```diff
@@ -6,13 +6,40 @@
 #include <Common/Exception.h>
 #include <Common/formatReadable.h>
 #include <common/logger_useful.h>
+#include <common/getThreadId.h>
 #include <Common/ProfileEvents.h>
 
 #include <atomic>
 #include <cmath>
 #include <random>
 #include <cstdlib>
 
+namespace
+{
+
+MemoryTracker * getMemoryTracker()
+{
+    if (auto * thread_memory_tracker = DB::CurrentThread::getMemoryTracker())
+        return thread_memory_tracker;
+
+    /// Do not use total_memory_tracker automatically in the main thread:
+    /// - there is not too much allocations in the main thread,
```

Comment:
\- there is not too much allocations
\+ there are not too many allocations

Reply:
Actually it will not be used there anyway, since it has it's own, so this comment will go away anyway